### PR TITLE
Solved Josephus Problem using recursion by @pshivesh8

### PR DESCRIPTION
**Pull request by pshivesh8**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Comment by pshivesh8** [comment]
Solved #2658 issue as a part of GSSOC'21

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Few changes: 
- Make the README updates 
- Add a function description

**Comment by pshivesh8** [comment]
I have updated the changes @HarshCasper please review it.

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -271,4 +271,5 @@ _add list here_
 - [Unique_Number_III](other/Unique_Number_III.cpp)
 - [KnapSack Fractional using Greedy Approach](other/knapsackfraction.cpp)
 - [Longest Subarray with equal number of 0 and 1](other/Longest_subarray_with_equal_0_1.cpp)
-- [Set Clear and Check bits](other/check_set_clear_bit.cpp)
+- [Josephus Problem](other/Josephus_Problem.cpp)
+- [Set Clear and Check bits](other/check_set_clear_bit.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Why is there a second Markdown update?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/README.md (hunk 1 above)
Comment (pshivesh8) [reply C-Plus-Plus/README.md]: Sir, I thought my readme file is not updated, so I re-committed it. Please consider the previous commit 00f08e3 , 87a989a .

**Review by pshivesh8 — COMMENTED** [review]
Please review

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Why did you delete the C++ README? Please revert that

**Pull request closed without merging**